### PR TITLE
Clear child browsing contexts on `Page.frameNavigated`

### PR DESCRIPTION
**Pull request by sadym-chromium**
Addressing the issue https://crbug.com/chromedriver/4204.
Whenever the page is navigated, the child browsing contexts should be removed.

WPT test `webdriver/tests/bidi/browsing_context/load/load.py` was disabled due to instability: https://github.com/web-platform-tests/wpt/issues/35846

### CODE REVIEW

**Review by thiagowfx — APPROVED** [review]

File: src/bidiMapper/domains/context/browsingContextProcessor.ts [hunk 1]
```diff
@@ -119,6 +146,30 @@ export class BrowsingContextProcessor {
       );
     });
 
+    sessionCdpClient.Page.on(
+      'lifecycleEvent',
```
Comment (OrKoN) [inline src/bidiMapper/domains/context/browsingContextProcessor.ts:150]: In Puppeteer, it's done on `frameNavigated`

Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextProcessor.ts]: Thanks! Changed to `Page.frameNavigated` as well.

File: src/bidiMapper/domains/context/browsingContextProcessor.ts [hunk 2]
```diff
@@ -38,7 +38,29 @@ export class BrowsingContextProcessor {
     );
   }
 
-  static #removeContext(contextId: string) {
+  async #removeContext(contextId: string): Promise<void> {
+    if (!BrowsingContextProcessor.#hasKnownContext(contextId)) {
+      return;
+    }
+    const context = BrowsingContextProcessor.#getKnownContext(contextId);
+    // Remove context's children.
```
Comment (OrKoN) [inline src/bidiMapper/domains/context/browsingContextProcessor.ts:46]: add `Context::removeChildren()` method and call it from here and frameNavigated?

File: src/bidiMapper/domains/context/browsingContextProcessor.ts [hunk 3]
```diff
@@ -38,15 +38,21 @@ export class BrowsingContextProcessor {
     );
   }
 
+  async #removeChildrenContexts(context: BrowsingContextImpl): Promise<void> {
+    // Remove context's children.
+    for (let child of context.children) {
+      await this.#removeContext(child.contextId);
```
Comment (OrKoN) [inline src/bidiMapper/domains/context/browsingContextProcessor.ts:44]: Is the order of remove important? Perhaps, `await Promise.all(context.children.map(child => this.#removeContext(child.contextId)))`?

**Review by OrKoN — APPROVED** [review]

File: src/bidiMapper/domains/context/browsingContextProcessor.ts [hunk 4]
```diff
@@ -38,7 +38,35 @@ export class BrowsingContextProcessor {
     );
   }
 
-  static #removeContext(contextId: string) {
+  async #removeChildrenContexts(context: BrowsingContextImpl): Promise<void> {
```
Comment (OrKoN) [inline src/bidiMapper/domains/context/browsingContextProcessor.ts:41]: nit: `removeChildContexts`

Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextProcessor.ts]: done

File: src/bidiMapper/domains/context/browsingContextProcessor.ts (hunk 3 above)
Comment (sadym-chromium) [reply src/bidiMapper/domains/context/browsingContextProcessor.ts]: done

**Pull request merged**